packages/generate-html/test/generate-html.spec.ts

```diff
@@ -73,19 +87,52 @@ describe('GenerateHTML', function () {
     ignoreDuplicateAttributes: false,
   };
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   testsMap.forEach((testSuite) => {
     const {
       certificatePath,
       expectedHtmlFromHbs,
       expectedHtmlFromMjml,
+      localOnly,
+      localTemplatePath,
+      localTemplatePartialPath,
       schemaExtraTranslationsPath,
       schemaTranslationsPath,
       type,
       version,
     } = testSuite;
     describe(`For ${type} - version ${version}`, () => {
+      let certificate: Record<string, unknown>;
+      let translations: Translations | undefined;
+      let extraTranslations: ExtraTranslations | undefined;
+      let templatePartial: HandlebarsTemplateDelegate<any> | undefined;
+      let localOnlyOptions: GenerateHtmlOptions | undefined;
```

Comment:
`localOnlyOptions` should be `extraOptions`, as you potentially always need to load the partials, not only when using local resources.

Reply:
The problem is that `localOnlyOptions` contains everything that is needed to stop network requests being made, whereas `extraOptions` is already present in some of the tests and only contains the properties necessary for a given test, such as on line 167. I added `localOnlyOptions` to differentiate between them easily. I've pushed some changes, if you are not happy with the refactor let me know and I will change it.

To avoid over-complicating the logic, I've added `handlebars: { partials: compiledPartials }` by default to the existing (default) `extraOptions` object, as when no partials are present it passes and empty object, which has no effect. I did this to avoid having multiple branches and improve readability (checking for the presence of `compiledPartials`, checking for `localOnly`, etc)

Reply:
Alright that makes sense and looks good.